File: shutdown.py
import os

def shutdown_computer():
    # Use system-specific commands to initiate the shutdown process
    try:
        if os.name == 'nt':  # For Windows
            os.system("shutdown /s /f /t 0")
        elif os.name == 'posix':  # For macOS and Linux
            os.system("sudo shutdown now")
        else:
            print("Unsupported operating system.")
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    shutdown_computer()

File: shutdown.cpp

#include <cstdlib> // For the system function

int main() {
    // Use system-specific commands to initiate the shutdown process
#ifdef _WIN32 // For Windows
    system("shutdown /s /f /t 0");
#elif __APPLE__ // For macOS
    system("sudo shutdown -h now");
#elif __linux__ // For Linux
    system("sudo shutdown now");
#else
    #error Unsupported operating system
#endif

    return 0;
}
